cpp/src/interop/from_arrow_host.cu

```diff
@@ -245,22 +245,49 @@ template <>
 std::unique_ptr<column> dispatch_copy_from_arrow_host::operator()<cudf::list_view>(
   ArrowSchemaView* schema, ArrowArray const* input, data_type type, bool skip_mask)
 {
-  void const* offset_buffers[2] = {nullptr, input->buffers[fixed_width_data_buffer_idx]};
-  ArrowArray offsets_array      = {
-         .length     = input->offset + input->length + 1,
-         .null_count = 0,
-         .offset     = 0,
-         .n_buffers  = 2,
-         .n_children = 0,
-         .buffers    = offset_buffers,
-  };
+  // Initialize schema for 32-bit ints regardless of list type
   nanoarrow::UniqueSchema offset_schema;
   NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(offset_schema.get(), NANOARROW_TYPE_INT32));
 
   ArrowSchemaView view;
   NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, offset_schema.get(), nullptr));
-  auto offsets_column =
-    this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+
+  size_type const physical_length = input->length + input->offset + 1;
```

Comment:
Would this overflow? Both `input->length` and `input->offset` can be `INT_MAX`.

Reply:
> Would this overflow? Both input->length and input->offset can be INT_MAX.

Good call, I added a check for it.

> Also, for my own understanding, shouldn't we just use the input->length as physical length and offset the input->buffers[fixed_width_data_buffer_idx] by input->offset?

For list types, I believe that engines are free to assume that the offset is specified only in the data child of list columns, and as a result offsetting only the offset child would not be legal. We might be able to make this work by changing the call to `get_column_copy(&view, input->children[0], child_type, false, stream, mr);` to replace `input->children[0]` with a new temporary ArrowArray with new offsets, but I didn't test it. I can give that a shot and see if I can find a way to do that safely.

Reply:
Thanks for explaining this. Yeah of course, that can be in a follow up PR, I was just curious

Reply:
Actually on second thought I don't think that we want to change this. In order to avoid copying the parts of the offsets buffer that are below the offset, we would have to additionally copy the data buffer, which we currently never have to do. The cost of that will probably outweigh the benefit of not copying the offsets unless the offset is pretty large (hard to be precise since it depends on what the data type is, e.g. if you have strings or nested types then the data could be skewed).